maggy/util.py

```diff
@@ -234,3 +229,48 @@ def _validate_ml_id(app_id, run_id):
     if prev_app_id == app_id and int(prev_run_id) >= run_id:
         return app_id, (int(prev_run_id) + 1)
     return app_id, run_id
+
+
+def _find_spark(conf=None):
+    """
+    Returns: SparkSession
+    """
+    # sp = SparkSession.builder.getOrCreate()
+    # sp.stop()
+
+    conf = SparkConf()
+    conf.set("num-executors", "1")
+
+    return (
+        SparkSession.builder.getOrCreate()
+        if not conf
```

Comment:
you always create the `SparkConf` so this will always be true. I don't think you should create a new sparksession with changed config on Hopsworks.

Reply:
this still bugs me @RiccardoGrigoletto  I don't see you anywhere using the `conf` argument